### PR TITLE
Implemented Next and Previous button for files preview

### LINKED ISSUE #8321: Add Next and Previous Scroll button for files preview
**Is your feature request related to a problem? Please describe.**
Currently the files preview only supports one file . There is no option to move to the next file preview or previous one. Only option is  close it and open the next one. Doctors in the field find its very difficult because often they upload lot of files and each time they need to open one by one. it takes lot of their time to assess patient files.

steps to reproduce
- Open a patient file
- Click on "files" from the patient dashboard
- upload some files and open one
- see the error

**Describe the solution you'd like**

- [ ] Avail a Next button or a " > "  on the right side and Previous button or " < " on the left.

If the first file is opened the previous button shouldn't be visible and if the last file is opened the Next button shouldn't be visible.

image

- [ ] Also increase the font size of the file name and add "Created on date and time" in fine text below the name. This would allow doctor to have more context on what image/document he/she is looking at. The text "File preview" may be reduced the font size.

**Additional context**
This feature will helps the doctors to Scroll through different files and assess the patients easily.

### PR DESCRIPTION
**Pull request by sarvesh-official**
# Add Next/Previous Navigation in File Preview Dialog

Fixes #8321

## Changes Made
- Added navigation buttons to allow users to move between files in the preview dialog
- Implemented proper file ordering with oldest files first and newest files last
- Added file creation date/time display for better context
- Improved file name display with larger font size
- Reduced "File Preview" title size for better visual hierarchy

## Implementation Details
1. **Navigation Controls**
   - Added Previous/Next buttons in the file preview dialog
   - Previous button shows for files after index 0
   - Next button shows for files before the last index
   - Buttons are hidden when no navigation is possible

2. **File Order & Display**
   - Reversed file array to show oldest files first
   - Added file creation timestamp below file name
   - Increased file name font size for better readability
   - Reduced modal title size

3. **State Management**
   - Added currentIndex tracking for proper navigation
   - Implemented proper index management during navigation
   - Added checks to prevent navigation beyond bounds

## Screenshots/Videos
image

https://github.com/user-attachments/assets/28648527-7533-4f22-ac45-5955a0230df7

## QA Checklist
- [ ] Test navigation between multiple files
- [ ] Verify correct file order display
- [ ] Check creation date/time accuracy
- [ ] Test edge cases (single file, first file, last file)
- [ ] Verify navigation button visibility logic
- [ ] Test with different file types (images, PDFs, etc.)

## Release Notes

- **New Features**
  - Enhanced file upload capabilities in the `FilePreviewDialog`, allowing users to navigate through multiple uploaded files with improved display of file information.
  - Added contextual handling of uploaded files with new properties in the `FileUpload` component.
  - Introduced keyboard navigation for file previewing, enhancing user interactivity.

- **Bug Fixes**
  - Minor adjustments to the `FileUploadPage` component for improved code cleanliness.

- **Documentation**
  - Updated interfaces to reflect new properties related to file uploads and management.

These changes aim to improve user experience by streamlining file handling and navigation within the application.

### CODE REVIEW

File: src/hooks/useFileManager.tsx [hunk 1]
```diff
@@ -72,6 +73,7 @@ export default function useFileManager(
   const [editDialogueOpen, setEditDialogueOpen] =
     useState<FileUploadModel | null>(null);
   const [editError, setEditError] = useState("");
+  const [currentIndex, setCurrentIndex] = useState<number>(-1);
```
[the remark this answers is not in the document]
Comment (sarvesh-official) [reply src/hooks/useFileManager.tsx:76]: This will reset the current index every time and raise uncontrolled issues.

**Comment by sarvesh-official** [comment]
Hi @rithviknishad, could you please review the PR as soon as possible? Let me know if it's good to merge, or feel free to suggest any changes. Thanks!

File: src/components/Common/FilePreviewDialog.tsx [hunk 2]
```diff
@@ -135,6 +190,14 @@ const FilePreviewDialog = (props: FilePreviewProps) => {
             </div>
           </div>
           <div className="flex flex-1 items-center justify-center">
+            {uploadedFiles && uploadedFiles.length > 1 && index > 0 && (
```
Comment (Jacobjeevan) [inline src/components/Common/FilePreviewDialog.tsx]: Let's remove the last condition; instead disable the button if index is 0, rather than hiding it.

Vice versa for the other button.

File: src/components/Common/FilePreviewDialog.tsx [hunk 3]
```diff
@@ -79,9 +101,27 @@ const FilePreviewDialog = (props: FilePreviewProps) => {
     });
   };
 
+  const handleNext = (newIndex: number) => {
+    if (!uploadedFiles?.length) return;
+    if (!loadFile) {
+      console.error("loadFile handler is not defined");
```
Comment (Jacobjeevan) [inline src/components/Common/FilePreviewDialog.tsx]: Remove the console log

File: src/components/Common/FilePreviewDialog.tsx [hunk 4]
```diff
@@ -135,6 +187,14 @@ const FilePreviewDialog = (props: FilePreviewProps) => {
...
           </div>
           <div className="flex flex-1 items-center justify-center">
+            {uploadedFiles && uploadedFiles.length > 1 && index > 0 && (
+              <ButtonV2
+                className="cursor-pointer bg-primary-500 rounded-md mr-2"
+                onClick={() => handleNext(index - 1)}
+              >
+                <CareIcon icon="l-arrow-left" className="h-4 w-4" />
+              </ButtonV2>
+            )}
```
[the remark this answers is not in the document]
Comment (Jacobjeevan) [reply src/components/Common/FilePreviewDialog.tsx]: Add the aria labels and attach arrow keys for onKeyDown respectively.

File: src/components/Common/FilePreviewDialog.tsx [hunk 5]
```diff
@@ -173,6 +236,17 @@ const FilePreviewDialog = (props: FilePreviewProps) => {
                 </div>
               )}
             </div>
+
+            {uploadedFiles &&
+              uploadedFiles.length > 1 &&
+              index < uploadedFiles.length - 1 && (
+                <ButtonV2
```
Comment (Jacobjeevan) [inline src/components/Common/FilePreviewDialog.tsx]: image

Adjust the margin b/w this button and the image.

**Review by Jacobjeevan — CHANGES_REQUESTED** [review]

File: src/components/Common/FilePreviewDialog.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (Jacobjeevan) [reply src/components/Common/FilePreviewDialog.tsx]: onKeyDown for Enter isn't needed.

File: src/components/Common/FilePreviewDialog.tsx (hunk 5 above)
Comment (sarvesh-official) [reply src/components/Common/FilePreviewDialog.tsx]: done

File: src/components/Common/FilePreviewDialog.tsx (hunk 3 above)
Comment (sarvesh-official) [reply src/components/Common/FilePreviewDialog.tsx]: done

File: src/components/Common/FilePreviewDialog.tsx (hunk 2 above)
Comment (sarvesh-official) [reply src/components/Common/FilePreviewDialog.tsx]: done

**Comment by sarvesh-official** [comment]
Hi @Jacobjeevan, I have updated the required changes, could you please review the PR again?

File: src/components/Common/FilePreviewDialog.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (sarvesh-official) [reply src/components/Common/FilePreviewDialog.tsx]: done

File: src/components/Common/FilePreviewDialog.tsx [hunk 6]
```diff
@@ -79,9 +101,23 @@ const FilePreviewDialog = (props: FilePreviewProps) => {
     });
   };
 
+  const handleNext = (newIndex: number) => {
+    if (!uploadedFiles?.length) return;
```
Comment (Jacobjeevan) [inline src/components/Common/FilePreviewDialog.tsx]: Let's combine these checks as well. Other than that, good to go.

Comment (sarvesh-official) [reply src/components/Common/FilePreviewDialog.tsx]: done

**Comment by sarvesh-official** [comment]
Hi @Jacobjeevan, I have updated the required changes, could you please review the PR again and merge it?

**Comment by nihal467** [comment]
@Jacobjeevan can you review it once and if the PR is ready for testing, mark it "needs testing"

**Review by Jacobjeevan — APPROVED** [review]

**Comment by nihal467** [comment]
LGTM

**Comment by sarvesh-official** [comment]
Hi @rithviknishad could you please review?

File: .prettierignore [hunk 7]
```diff
@@ -4,4 +4,4 @@ lib
 build
 *.css
 *.gen.tsx
-*.bs.js
+*.bs.js
```
Comment (rithviknishad) [inline .prettierignore]: There are unintended line ending changes unrelated to this PR. Refer #9223 on how to revert these line ending changes.

**Review by rithviknishad — CHANGES_REQUESTED** [review]

**Review by rithviknishad — APPROVED** [review]

**Comment by sarvesh-official** [comment]
Hi @rithviknishad, do I need to make any changes, or is this PR good to merge?
